### PR TITLE
Select repoid for publish to debian repo

### PR DESCRIPTION
**Pull request by karajas**
Passing in debian keys in pipebuild to allow selection of keys used to publish to the debian repo

/cc @chcosta @rakeshsinghranchi @eerhardt

### CODE REVIEW

File: publish/publish.proj [hunk 1]
```diff
@@ -53,6 +53,26 @@
...
+        </RepoIds>
+        <RepoIds Include="DebianId_ubuntu1604-x64">
+          <Key>$(DebianId_ubuntu1604-x64)</Key>
+        </RepoIds>
+        <RepoIds Include="DebianId_ubuntu1610-x64">
+          <Key>$(DebianId_ubuntu1610-x64)</Key>
+        </RepoIds>
+      </ItemGroup>
+      <PropertyGroup>
+        <RidWithoutDots>$([System.String]::Copy($(DistroRid)).Replace('.', ''))</RidWithoutDots>
```
Comment (chcosta) [inline publish/publish.proj]: Why not leave the dots?

Comment (karajas) [inline publish/publish.proj]: Since the keys themselves come from PropertyGroups, it looks like those can't have dots

**Comment by chcosta** [comment]
I imagine the keys aren't mentioned here, because they will be stored in the VSTS PipeBuild definition.  What about the other values?  Will that be a separate PR?

**Comment by karajas** [comment]
I'll add the other keys in the build definition now.

File: publish/publish.proj (hunk 1 above)
Comment (chcosta) [inline publish/publish.proj]: I see, yes, it's an invalid character.

```
error MSB5016: The name "Hello.1.2.3" contains an invalid character "."
```

File: publish/publish.proj [hunk 2]
```diff
@@ -53,6 +53,26 @@
...
+        <DebRepoId Condition="$([System.String]::Copy(%(RepoIds.Identity)).EndsWith('$(RidWithoutDots)'))">%(RepoIds.Key)</DebRepoId>
+      </PropertyGroup>
+      <Message Importance="High" Condition="'$(DebRepoId)'=='' OR '$(DebRepoUser)'=='' OR '$(DebRepoPass)'=='' OR '$(DebRepoServer)'==''"
```
Comment (chcosta) [inline publish/publish.proj]: This should probably be an error

**Comment by karajas** [comment]
@chcosta Does the debian 8 rid look correct to you?

**Comment by karajas** [comment]
Verified that debian 8 is correct by checking the VERSION_ID on the docker image.

**Comment by rakeshsinghranchi** [comment]
@karajas ,Have you manually verified the push by looking at the debian feed for all Ubuntu and debian versions?

**Comment by karajas** [comment]
I pulled down the docker image yesterday but hit an issue during the normal build. 

```
/newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: 1: /newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: /LICENSE.txt: not found
  /newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: 2: /newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: /ThirdPartyNotices.txt: not found
  /newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: 3: /newlinux/core-setup/Bin/obj/ubuntu.14.04-x64.Release/packages/dotnet-host/2.0.0-preview2-25309-9/dotnet-host-2.0.0-preview2-25309-9/debian/install: /dotnet: not found
  dh_install: problem reading debian/install:
  make: *** [binary] Error 127
dpkg-buildpackage : error : fakeroot debian/rules binary gave error exit status 2 [/newlinux/core-setup/src/pkg/packaging/dir.proj]
  debuild: fatal error at line 1364:
  dpkg-buildpackage -rfakeroot -D -us -uc failed
```

**Comment by rakeshsinghranchi** [comment]
cc @eerhardt , @chcosta 
Chris had hit this issue as well.

**Comment by karajas** [comment]
I think this is an issue with being able to run as root on docker.

**Comment by karajas** [comment]
Got this working now.

**Comment by karajas** [comment]
```
Adding package to azure-apt-cat.cloudapp.net []
    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                   Dload  Upload   Total   Spent    Left  Speed

    0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  HTTP/1.1 202 Accepted
  X-Powered-By: Express
  Location: /v1/packages/queue/5913a8bf41e38e177a99add9
  Date: Wed, 10 May 2017 23:56:47 GMT
  Connection: keep-alive
  Content-Length: 0
```
Was the response I got

**Comment by karajas** [comment]
@dotnet-bot test Ubuntu16.04 x64 Release Build please

File: publish/dir.targets [hunk 3]
```diff
@@ -30,13 +32,13 @@
   <Target Name="GenerateDebRepoFiles"
           Condition="('$(TargetsUbuntu)' == 'true' or '$(TargetsDebian)' == 'true') and '$(TargetArchitecture)' == 'x64'">
     <ItemGroup>
-      <DebInstallerFile Include="$(PackagesOutDir)dotnet-host-*$(SharedFrameworkNugetVersion).deb" >
+      <DebInstallerFile Include="$(PackagesOutDir)dotnet-host-$(DistroRid)-$(SharedFrameworkNugetVersion).deb" >
         <PackageName>dotnet-host</PackageName>
```
Comment (eerhardt) [inline publish/dir.targets]: I think you should reuse these properties for this:
[https://github.com/dotnet/core-setup/blob/c54c12905b41cd5257794a2e06f7c9b5a1dc5fe4/src/pkg/packaging/dir.props#L29-L31](https://github.com/dotnet/core-setup/blob/c54c12905b41cd5257794a2e06f7c9b5a1dc5fe4/src/pkg/packaging/dir.props#L29-L31).

You may need to move them higher, or import the packaging/dir.props files.  Or refactor them out into a shared .props file.

Also, there seems to be a change here.  In the packaging props file, we use a `.` in between RID and Version.  Here you are using a `-`.

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 4]
```diff
@@ -374,6 +374,15 @@
     },
     "PB_PortableBuild": {
       "value": "false"
+    },
+    "PB_DebRepoUser": {
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: I don't see this variable being used.

File: buildpipeline/Core-Setup-Linux-BT.json [hunk 5]
```diff
@@ -177,7 +177,7 @@
       },
       "inputs": {
         "filename": "docker",
-        "arguments": "run --rm $(DockerCommonRunArgs) $(PB_GitDirectory)/Tools/msbuild.sh $(PB_GitDirectory)/publish/publish.proj /p:AzureAccountName=$(PB_AzureAccountName) /p:AzureAccessToken=$(PB_AzureAccessToken) /p:DistroRid=$(PB_DistroRid) /p:TargetArchitecture=$(PB_TargetArchitecture) /p:ConfigurationGroup=$(BuildConfiguration) /p:PortableBuild=$(PB_PortableBuild) /p:OSGroup=Linux",
+        "arguments": "run --rm $(DockerCommonRunArgs) $(PB_GitDirectory)/Tools/msbuild.sh $(PB_GitDirectory)/publish/publish.proj /p:AzureAccountName=$(PB_AzureAccountName) /p:AzureAccessToken=$(PB_AzureAccessToken) /p:DistroRid=$(PB_DistroRid) /p:TargetArchitecture=$(PB_TargetArchitecture) /p:ConfigurationGroup=$(BuildConfiguration) /p:PortableBuild=$(PB_PortableBuild) /p:OSGroup=Linux /p:DebRepoUser=$(PB_DistroRid) /p:DebRepoServer=$(PB_DebRepoServer) /p:DebRepoPass=$(DEB_REPO_PASSWORD) $(PB_DebianKeys)",
```
Comment (eerhardt) [inline buildpipeline/Core-Setup-Linux-BT.json]: `/p:DebRepoUser=$(PB_DistroRid)` this doesn't seem right to me.

**Comment by eerhardt** [comment]
> if [[ "$1" == "-listrepos" ]]; then

Doesn't this need to change to `$5`?

I'm not sure I understand how the environment variables are being set with your change.  Can you explain how they are getting set?

---
Refers to: tools-local/scripts/publish/repoapi_client.sh:168 in 8230a8d. [](commit_id = 8230a8dbb4da222e9947dd38df60d00b00950cf0, deletion_comment = False)

**Comment by eerhardt** [comment]
> #   $REPO_SERVER

If we are no longer using environment variables, this comment should be updated.

---
Refers to: tools-local/scripts/publish/repoapi_client.sh:10 in 8230a8d. [](commit_id = 8230a8dbb4da222e9947dd38df60d00b00950cf0, deletion_comment = False)

**Pull request merged**